### PR TITLE
LOG-5820: Refactor filter validation to OBS

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR: 
* refactors the filter validations to the observability API
* Adds declarative API to block upon creation

### Links
* https://issues.redhat.com/browse/LOG-5820
* fixes https://issues.redhat.com/browse/LOG-5851
* fixes https://issues.redhat.com/browse/LOG-5852

cc @Clee2691 @cahartma @vparfonov

### CODE REVIEW

File: internal/validations/observability/filters/validate_filters.go [hunk 1]
```diff
@@ -18,38 +18,33 @@ var (
...
-	for _, filterSpec := range clf.Spec.Filters {
-		switch filterSpec.Type {
-		case loggingv1.FilterDrop:
-			validateDropFilter(&filterSpec, &clf.Status.Filters)
-		case loggingv1.FilterPrune:
-			validatePruneFilter(&filterSpec, &clf.Status.Filters)
-		}
+	results := []string{}
+	switch spec.Type {
+	case loggingv1.FilterDrop:
```
Comment (cahartma) [inline internal/validations/observability/filters/validate_filters.go:25]: obs

Comment (cahartma) [reply internal/validations/observability/filters/validate_filters.go:25]: and the line below btw

**Comment by cahartma** [comment]
@jcantrill okay to override this test failure?

**Comment by jcantrill** [comment]
@cahartma updated to resolve a CRD install issue

/hold cancel

**Pull request merged**